buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -428,14 +428,27 @@ public synchronized void start() {
 
         createConfiguration();
 
+        final List<String> pluginsToInstall = new ArrayList<>();
```

Comment:
Is this intermediary collection really necessary? All it's doing is saving us a single `toString()` call later on.

I mainly say that as it's a bit confusing. Having a separate "plugins to install" to me implies that `this.plugins` is some kind of superset of that, which isn't the case. We always install all configured plugins for the cluster.

Reply:
Note that we sometimes add `x-pack` to the list. However, we can remove that code in master. Let me do that in a follow-up, and then we can remove the extra collection then, too.

Reply:
I opened #51441 for this.